FF1Lib/Armor.cs

```diff
@@ -81,6 +92,362 @@ public void ExpandArmor()
+
+		    var spellHelper = new SpellHelper(this);
+		    var allSpells = GetSpells();
+		    var generatedItems = new HashSet<FF1Lib.Item>();
+		    var generatedNames = new HashSet<string>();
+		    for (int tier = 2; tier >= 0; tier--) {
+			List<int> requireType;
+			List<int> requireClasses;
+			switch(tier) {
+			    case 0:
+				requireType = new List<int>    {BRACELET, BRACELET, BRACELET, ARMOR, ARMOR};
+				requireClasses = new List<int> {   LIGHT,   MEDIUM,    HEAVY, MEDIUM, HEAVY};
+				break;
+			    case 1:
+				requireType = new List<int>();
+				requireClasses = new List<int>();
+				break;
+			    case 2:
+				requireType = new List<int>    { ARMOR, SHIELD,   HELM, GAUNTLET};
+				requireClasses = new List<int> {KNIGHT, KNIGHT, KNIGHT,   KNIGHT};
+				break;
+			    default:
+				requireType = new List<int>();
+				requireClasses = new List<int>();
+				break;
+			}
+			for (int count = 0; count < tierList[tier].Count; ) {
+			    var itemId = tierList[tier][count];
+
+			    if (generatedItems.Contains(itemId)) {
+				count++;
+				continue;
+			    }
+
+			    int armorType;
+			    int armorClass;
+
+			    // incentive armor that needs to be re-created:
+			    //
+			    // opal bracelet
+			    // power bonk
+			    // white shirt
+			    // black shirt
+			    // ribbon
+
+			    var accessories = new List<int> {HELM, CAPE, RING};
+
+			    armorType = rng.Between(0, 7);
+			    armorClass = rng.Between(0, 3);
+			    if (itemId == Item.Opal) {
+				armorType = BRACELET;
+				armorClass = KNIGHT;
+			    } else if (itemId == Item.PowerGauntlets) {
+				armorType = GAUNTLET;
+			    } else if (itemId == Item.WhiteShirt) {
+				armorType = SHIRT;
+				armorClass = HEAVY;
+			    } else if (itemId == Item.BlackShirt) {
+				armorType = SHIRT;
+				armorClass = HEAVY;
+			    } else if (itemId == Item.Ribbon) {
+				armorType = accessories[rng.Between(0, accessories.Count-1)];
+				armorClass = CLOTH;
+			    } else if (requireType.Count > 0) {
+				armorType = requireType[0];
+				armorClass = requireClasses[0];
+				requireType.RemoveAt(0);
+				requireClasses.RemoveAt(0);
+			    } else if (armorType == SHIRT || armorType == BRACELET) {
+				// Don't roll tier 0 shirts or any extra bracelets
+				if (tier == 0 || armorType == BRACELET) {
+				    while (armorType == SHIRT || armorType == BRACELET) {
+					armorType = rng.Between(0, 7);
+				    }
+				}
+			    }
+
+			    var name = "";
+			    byte weight = 0;
+			    byte absorb = 0;
+			    byte elementalResist = 0;
+			    byte spellIndex = 0xFF;
+			    ArmorType type = ArmorType.ARMOR;
+			    ushort permissions = 0;
+
+			    absorb = (byte)RangeScale(armorDefenseBase[armorClass, armorType], .7, 1.4, 1.0, rng);
+			    weight = (byte)RangeScale(armorEvadePenaltyBase[armorClass, armorType], .7, 1.4, 1.0, rng);
+
+			    int nameClass = armorClass;
+			    name = classNames[nameClass][rng.Between(0, classNames[nameClass].Length-1)];
+			    switch (armorClass) {

```

Comment:
Just a consideration. Perhaps you want to play around a little more with permissions here. Like restrict the cloth class to mages...

Reply:
Shirts (robes) are already restricted to mages, but can roll higher quality than CLOTH.

Reply:
ok